library/include/er/stack_alloc.h

```diff
@@ -0,0 +1,68 @@
+#include <memory>
+#include <new>
+#include <algorithm>
+
+namespace er {
+
+template <typename T, size_t size>
+class StackAlloc {
+public:
+    using value_type = T;
+    using pointer = T*;
+    using const_pointer = const T*;
+    using reference = T&;
+    using const_reference = const T&;
+    using size_type = size_t;
+    using difference_type = ptrdiff_t;
+
+    pointer allocate(size_type n) {
+        if (sizeof(T) > size) {
+            return static_cast<pointer>(::operator new(n * sizeof(T)));
+        } else {
+            return static_cast<pointer>(::operator new(n * sizeof(T), std::nothrow));
```

Comment:
It's actually doesn't allocate memory on stack, it's do it on heap. Just reinterpret_cast _stack and return pointer